src/garage/np/algos/base.py

```diff
@@ -25,4 +30,60 @@ def train(self, runner):
             The average return in last epoch cycle or None.
 
         """
-        pass
+
+    def evaluate_performance(self, itr, batch):
+        # pylint: disable=no-self-use
+        r"""Evaluate the performance of the algorithm.
+
+        Args:
+            itr (int): Iteration number.
+            batch (dict): Evaluation trajectories, representing
+                the best current performance of the algorithm, with keys:
+                * env_spec (garage.envs.EnvSpec): Specification for the
+                environment from which this data was sampled.
+                * observations (numpy.ndarray): A numpy array containing the
+                    observations for all time steps in this batch.
+                * actions (numpy.ndarray): A  numpy array containing the
+                    actions for all time steps in this batch.
+                * rewards (numpy.ndarray): A numpy array containing the
+                    rewards for all time steps in this batch.
+                * terminals (numpy.ndarray): A boolean numpy array
+                    containing the termination signals for all time steps
+                    in this batch.
+                * env_infos (dict): A dict of numpy arrays arbitrary
+                    environment state information.
+                * agent_infos (numpy.ndarray): A dict of numpy arrays
+                    arbitrary agent state information.
+                * lengths (numpy.ndarray): An integer numpy array
+                    containing the length of each trajectory in this batch.
+                * discount (float): Discount value.
+                * episode_reward_mean (collections.deque): Average episode
```

Comment:
this should be calculated from the `rewards` data

Reply:
I'm trying to avoid using any self variable (in this case `self.episode_reward_mean`) in the interface. So I passed this from the batch.

Reply:
I see, you don't want to change how EpisodeRewardMean is calculated.

In that case, just move logging it back into the algorithm and log it under `Extras/EpisodeRewardMean` as before. This metric is only used for comparing to some openai/baselines results, anyway.